src/mrpro/operators/FourierOp.py

```diff
@@ -87,36 +85,16 @@ def get_traj(traj: KTrajectory, dims: Sequence[int]):
                     'k-space dimension, i.e. kx along k0, ky along k1 and kz along k2',
                 )
 
-            self._nufft_im_size = get_spatial_dims(recon_matrix, self._nufft_dims)
-            grid_size = [int(size * nufft_oversampling) for size in self._nufft_im_size]
-            omega = [
-                k * 2 * torch.pi / ks
-                for k, ks in zip(
-                    get_traj(traj, self._nufft_dims),
-                    get_spatial_dims(encoding_matrix, self._nufft_dims),
-                    strict=True,
-                )
-            ]
-
-            # Broadcast shapes not always needed but also does not hurt
-            omega = [k.expand(*np.broadcast_shapes(*[k.shape for k in omega])) for k in omega]
-            self.register_buffer('_omega', torch.stack(omega, dim=-4))  # use the 'coil' dim for the direction
-
-            self._fwd_nufft_op = KbNufft(
-                im_size=self._nufft_im_size,
-                grid_size=grid_size,
-                numpoints=nufft_numpoints,
-                kbwidth=nufft_kbwidth,
-            )
-            self._adj_nufft_op = KbNufftAdjoint(
-                im_size=self._nufft_im_size,
-                grid_size=grid_size,
-                numpoints=nufft_numpoints,
-                kbwidth=nufft_kbwidth,
+            self._non_uniform_fast_fourier_op = NonUniformFastFourierOp(
+                dim=tuple(self._nufft_dims),
+                recon_matrix=get_spatial_dims(recon_matrix, self._nufft_dims),
+                encoding_matrix=get_spatial_dims(encoding_matrix, self._nufft_dims),
+                traj=traj,
+                nufft_oversampling=nufft_oversampling,
+                nufft_numpoints=nufft_numpoints,
+                nufft_kbwidth=nufft_kbwidth,
```

Comment:
I would remove 
                nufft_numpoints
                nufft_kbwidth
from Fourierop.

It doesn't make sense to change the kbwidth, the number of points seems to be an implementation detail of some nufft implementations, but is not generally available in all. Also, there is no guidance how to choose.

I would only keep the oversampling as the main performance knob and mention in the docstring that it can be lowered to 1.25 to improve performance at the cost of qualitiy.